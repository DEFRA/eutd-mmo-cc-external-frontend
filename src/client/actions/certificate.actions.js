import { isError, showFullPageError } from './index';

export const GET_SUMMARY_DOCUMENT = 'GET_SUMMARY_DOCUMENT';
export const SHOW_SUMMARY_DOCUMENT_ERRORS = 'SHOW_SUMMARY_DOCUMENT_ERRORS';
export const GET_SUMMARY_UNAUTHORISED = 'GET_SUMMARY_DOCUMENT_UNAUTHORISED';

export function showSummaryCertificateErrors(data) {
  return { type: SHOW_SUMMARY_DOCUMENT_ERRORS, payload: { validationErrors: data } };
}

export const getSummaryCertificate = (journey, documentNumber = undefined) => async (dispatch, getState, { orchestrationApi }) => {
  const config = (documentNumber)
    ? { headers: { documentnumber: documentNumber } }
    : {};

  await orchestrationApi.get(`/certificate/${journey}`, config)
    .then(
      res => {
        dispatch({ type: GET_SUMMARY_DOCUMENT, payload: res.data });
      },
      error => {
        if (isError(error)) {
          dispatch(showFullPageError());
        } else {
          dispatch(unauthorised());
        }
      });

    function unauthorised() { return { type: GET_SUMMARY_UNAUTHORISED }; }
};