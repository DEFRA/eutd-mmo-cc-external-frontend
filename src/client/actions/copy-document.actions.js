import { showFullPageError, apiCallFailed, isError, CLEAR_ERRORS } from './index';
export const CONFIRM_COPY_DOCUMENT            = 'confirm_copy_document';
export const UNAUTHORISED_COPY_DOCUMENT       = 'unauthorised_copy_document';
export const CLEAR_COPY_DOCUMENT              = 'clear_copy_document';
export const REMOVE_CONFIRM_COPY_CERTIFICATE  = 'remove_confirm_copy_certificate';
export const ADD_ERROR_COPY_CERTIFICATE       = 'add_error_copy_certificate';
export const REMOVE_ERROR_COPY_CERTIFICATE    = 'remove_error_copy_certificate';

export const submitCopyCertificate = (payload) => async (dispatch, _getState, {orchestrationApi}) => {
  const {documentNumber, copyDocumentAcknowledged, voidOriginal, excludeLandings, journey} = payload;
  const config = {headers: {documentnumber: documentNumber}};

  try {
    let ipAddress;
    if (voidOriginal) {
      const ip = await orchestrationApi.get('/client-ip');
      ipAddress = ip.data;
    }

    return await orchestrationApi.post('/confirm-copy-certificate', { copyDocumentAcknowledged, voidOriginal, excludeLandings, journey, ipAddress }, config)
    .then(res => {
      dispatch({ type: CONFIRM_COPY_DOCUMENT, payload: {
        documentNumber: res.data.documentNumber,
        copyDocumentAcknowledged: res.data.copyDocumentAcknowledged,
        voidDocumentConfirm: res.data.voidOriginal
      }});
      return res.data.newDocumentNumber;
    });
  } catch (e) {
    if (isError(e)) dispatch(showFullPageError());
    if (e.response && e.response.status === 403) {
      dispatch({ type: UNAUTHORISED_COPY_DOCUMENT }); return;
    } else {
      dispatch(apiCallFailed(e.response));
    }
  }
};

export const checkCopyCertificate = (documentNumber) => async (dispatch, _getState, {orchestrationApi}) => {
  const config = {
    headers: {
      documentnumber: documentNumber
    }
  };

  try {
    const res = await orchestrationApi.get('/check-copy-certificate', config);
    const { canCopy } = res.data;
    if (!canCopy) {
      dispatch(unauthorised());
    }
  } catch(e) {
    if (isError(e)) dispatch(showFullPageError());
    if (e.response && e.response.status === 403) {
      dispatch(unauthorised());
      throw new Error(`Unauthorised access ${e.response.status}`);
    } else {
      dispatch(apiCallFailed(e.response));
      throw new Error(`An error has occurred ${e.response.status} (${e.response.statusText})`);
    }
  }

  function unauthorised() { return { type: UNAUTHORISED_COPY_DOCUMENT }; }
};

export const addCopyDocument = (data) => (dispatch) => dispatch({ type: CONFIRM_COPY_DOCUMENT, payload: data });
export const clearErrors = () => (dispatch) =>  dispatch({ type: CLEAR_ERRORS });
export const clearCopyDocument = () => (dispatch) => dispatch({ type: CLEAR_COPY_DOCUMENT });
export const unauthorisedCopyDocument = () => (dispatch) => dispatch({ type: UNAUTHORISED_COPY_DOCUMENT });