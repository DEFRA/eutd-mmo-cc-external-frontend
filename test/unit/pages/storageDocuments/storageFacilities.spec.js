import storageFacilities from '../../../../src/client/pages/storageNotes/storageFacilitiesPage';
import {
  getStorageNotesFromRedis,
  saveStorageNotes,
  clearStorageNotes
} from '../../../../src/client/actions';
import { clearPostcodeLookupAddress } from '../../../../src/client/actions/postcode-lookup.actions';
import * as React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { component as StorageFacilitiesPage } from '../../../../src/client/pages/storageNotes/storageFacilitiesPage';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { render } from '@testing-library/react';

jest.mock('../../../../src/client/actions');
jest.mock('../../../../src/client/actions/postcode-lookup.actions');

describe('loadData', () => {

  const store = {
    dispatch: () => {
      return new Promise((res) => {
        res();
      });
    }
  };

  const journey = 'storageDocument';
  const documentNumber = 'some-document-number';

  beforeEach(() => {
    getStorageNotesFromRedis.mockReturnValue({ type: 'GET_STORAGE_NOTES_FROM_REDIS' });
  });

  it('will call all methods needed to load the component', async () => {
    storageFacilities.documentNumber = documentNumber;

    await storageFacilities.loadData(store, journey);

    expect(getStorageNotesFromRedis).toHaveBeenCalledWith(documentNumber);
  });

});

describe('component', () => {

  const windowScrollTo = window.scrollTo = jest.fn();

  const mockStore = configureStore([thunk]);

  const initialState = {
    storageNotes: {
      storageFacilities: [
        {
          facilityName: 'Facility 1'
        },
        {
          facilityName: 'Facility 2'
        }
      ]
    }
  };

  const store = mockStore(initialState);

  const props = {
    route: {
      journey: 'storageNotes',
      progressUri: '/create-storage-document/progress'
    },
    history: {
      location: {
        pathname: '/'
      }
    }
  };
  
  const wrapper = mount(
    <Provider store={store}>
      <MemoryRouter>
        <StorageFacilitiesPage {...props} />
      </MemoryRouter>
    </Provider>
  );

  beforeEach(() => {
    saveStorageNotes.mockReset();
    saveStorageNotes.mockReturnValue({ type: 'SAVE' });

    clearPostcodeLookupAddress.mockReset();
    clearPostcodeLookupAddress.mockReturnValue({ type: 'CLEAR' });
    clearStorageNotes.mockReturnValue({type: 'CLEAR_STORAGE_NOTES'});
  });


  it('should take a snapshot of Storage Facility', ()=> {
    const { container } = render(wrapper);
    expect(container).toMatchSnapshot();
  });

  it('will render', () => {
    expect(wrapper).toBeDefined();
  });

  it('will scroll to 0,0', () => {
    expect(windowScrollTo).toHaveBeenCalledWith(0, 0);
  });

  it('will call getFromRedis', () => {
    expect(getStorageNotesFromRedis).toHaveBeenCalled();
  });

  it('will render a delete button for each facility', () => {
    expect(wrapper.exists('#remove-facility-0')).toBe(true);
    expect(wrapper.exists('#remove-facility-1')).toBe(true);
  });

  it('should have a back to the progress page link', () => {
    expect(wrapper.find('a[href="/create-storage-document/progress"]').exists()).toBeTruthy();
  });

  it('will handle removing a storage facility', () => {
    wrapper.find('#remove-facility-0').at(0).simulate('click');

    expect(saveStorageNotes).toHaveBeenCalledWith(
      {
        storageFacilities: [
          {
            facilityName: 'Facility 2'
          }
        ]
      }
    );

    expect(clearPostcodeLookupAddress).toHaveBeenCalled();
  });


  it('should NOT push history for componentDidMount when hasRequiredData is true', async () => {
    const mockPush = jest.fn();

    const props = {
      history: {
        push: mockPush,
      },
      match: {
        params: {
          documentNumber: 'some-doc-num'
        }
      },
      storageNotes: {
        storageFacilities: [
          {
            facilityName: 'Facility 1',
          },
          {
            facilityName: 'Facility 2',
          }
        ]
      },
      getFromRedis: jest.fn(),
      getAllFish: jest.fn()
    };

    props.history.push.mockReset();

    await new StorageFacilitiesPage.WrappedComponent(props).componentDidMount();

    expect(mockPush).not.toHaveBeenCalled();
  });

  it('should push history for componentDidMount when hasRequiredData is false', async () => {
    const mockPush = jest.fn();

    const props = {
      match: {
        params: {
          documentNumber: 'some-doc-num'
        }
      },
      history: {
        push: mockPush,
      },
      storageNotes: {},
      getFromRedis: jest.fn(),
      getAllFish: jest.fn()
    };

    props.history.push.mockReset();

    await new StorageFacilitiesPage.WrappedComponent(props).componentDidMount();

    expect(mockPush).toHaveBeenCalledWith('/create-storage-document/some-doc-num/add-storage-facility-details');
  });

  describe('if no storage facilities have been updated', () => {

    const initialState = {
      storageNotes: {
        storageFacilities: [
          {
            facilityName: 'Facility 1',
            _facilityUpdated: false
          },
          {
            facilityName: 'Facility 2',
            _facilityUpdated: false
          }
        ]
      }
    };

    const store = mockStore(initialState);

    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <StorageFacilitiesPage {...props} />
        </MemoryRouter>
      </Provider>
    );

    it('will not show the notification banner', () => {
      expect(wrapper.exists('NotificationBanner')).toBe(false);
    });

  });

  describe('if any storage facilities have been updated', () => {

    const initialState = {
      storageNotes: {
        storageFacilities: [
          {
            facilityName: 'Facility 1',
            _facilityUpdated: false
          },
          {
            facilityName: 'Facility 2',
            _facilityUpdated: true
          }
        ]
      }
    };

    const store = mockStore(initialState);

    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <StorageFacilitiesPage {...props} />
        </MemoryRouter>
      </Provider>
    );

    it('will show the notification banner', () => {
      expect(wrapper.exists('NotificationBanner')).toBe(true);
    });

  });

  describe('if we have any errors', () => {

    const initialState = {
      storageNotes: {
        storageFacilities: [
          {
            facilityName: 'Facility 1'
          },
          {
            facilityName: 'Facility 2'
          },
          {
            facilityName: 'Facility 3'
          }
        ],
        errors: {
          'addAnotherStorageFacility': 'Error one',
          'storageFacilities-0-facilityAddressOne': 'Error two',
          'storageFacilities-1-facilityAddressOne': 'Error three'
        },
        errorsUrl: '/'
      }
    };


    const store = mockStore(initialState);

    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <StorageFacilitiesPage {...props} />
        </MemoryRouter>
      </Provider>
    );

    it('will show an error island', () => {
      expect(wrapper.exists('#errorIsland')).toBe(true);
    });

    it('will list all errors in the error island', () => {
      const errors = wrapper.find('#errorIsland li');

      expect(errors.length).toBe(3);
      expect(errors.at(0).html()).toContain('Error one');
      expect(errors.at(1).html()).toContain('Error two');
      expect(errors.at(2).html()).toContain('Error three');
    });

    it('will update the links for facility errors to highlight the row which has the error', () => {
      const errors = wrapper.find('#errorIsland li');

      expect(errors.at(1).html()).toContain('href="#storageFacilities-0"');
      expect(errors.at(2).html()).toContain('href="#storageFacilities-1"');
    });

    it('will not update the link for any other errors', () => {
      const errors = wrapper.find('#errorIsland li');

      expect(errors.at(0).html()).toContain('href="#addAnotherStorageFacility"');
    });

    it('will show inline errors for storage facilities with errors', () => {
      expect(wrapper.find('#storageFacilities-0').exists('.form-group-error')).toBe(true);
      expect(wrapper.find('#storageFacilities-0').exists('.error-message')).toBe(true);
      expect(wrapper.find('#storageFacilities-0 .error-message').html()).toContain('Error two');

      expect(wrapper.find('#storageFacilities-1').exists('.form-group-error')).toBe(true);
      expect(wrapper.find('#storageFacilities-1').exists('.error-message')).toBe(true);
      expect(wrapper.find('#storageFacilities-1 .error-message').html()).toContain('Error three');
    });

    it('will not show inline errors for storage facilities without errors', () => {
      expect(wrapper.find('#storageFacilities-2').exists('.form-group-error')).toBe(false);
      expect(wrapper.find('#storageFacilities-2').exists('.error-message')).toBe(false);
    });

  });
});

describe('should redirect to the forbidden page', () => {
  const mockPush = jest.fn();
  const mockGetFromRedis = jest.fn();

  it('should redirect to the forbidden page if storageNotes is unauthorised', async () => {
    const props = {
      match: {
        params: { documentNumber: 'GBR-23423-4234234' }
      },
      global: {
        allFish: []
      },
      history: {
        push: mockPush,
      },
      exporter: {
        model: {
          exporterFullName: 'Mr Fish Exporter',
          exportCompanyName: 'MailOrderFish Ltd',
          addressOne: '5 Costa Road',
          townCity: 'Hackney',
          postcode: 'NE1 4FS',
          _updated: true
        }
      },
      storageNotes: {
        unauthorised: true
      },
      transport: {
        vehicle: 'plane',
        exportedTo: {
          isoCodeAlpha2: 'Alpha2',
          isoCodeAlpha3: 'Alpha3',
          isoNumericCode: 'IsoNumericCode',
          officialCountryName: 'Eritrea'
        }
      },
      getFromRedis: mockGetFromRedis
    };

    await new StorageFacilitiesPage.WrappedComponent(props).componentDidUpdate();

    expect(mockPush).toHaveBeenCalledWith('/forbidden');
  });
});

describe('should take snapshot of the page based on number of facilities added', () => {
  const mockStore = configureStore([thunk]);
  const mockClearErrors = jest.fn();


  clearPostcodeLookupAddress.mockReturnValue({ type: 'CLEAR' });
  clearStorageNotes.mockReturnValue({type: 'CLEAR_STORAGE_NOTES'});

  const props = {
    clear: mockClearErrors,
    route: {
      journey: 'storageNotes',
      progressUri: '/create-storage-document/progress'
    },
    history: {
      location: {
        pathname: '/'
      }
    }
  };
    
    it('should take a snapshot of Storage Facility when only 1 facility added', ()=> {
      const initialState = {
        storageNotes: {
          storageFacilities: [
            {
              facilityName: 'Facility 1'
            }
          ]
        }
      };
    
      const store = mockStore(initialState);
      const wrapper = mount(
        <Provider store={store}>
          <MemoryRouter>
            <StorageFacilitiesPage {...props} />
          </MemoryRouter>
        </Provider>
      );
      const { container } = render(wrapper);
      expect(container).toMatchSnapshot();
    });

    it('should take a snapshot of Storage Facility when no storage facilities added', ()=> {
      const initialState = {
        storageNotes: {
          storageFacilities: [
          ]
        }
      };
    
      const store = mockStore(initialState);
      const wrapper = mount(
        <Provider store={store}>
          <MemoryRouter>
            <StorageFacilitiesPage {...props} />
          </MemoryRouter>
        </Provider>
      );
      const { container } = render(wrapper);
      expect(container).toMatchSnapshot();
    });
});