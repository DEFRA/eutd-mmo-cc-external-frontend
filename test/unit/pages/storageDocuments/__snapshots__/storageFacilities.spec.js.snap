// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component should take a snapshot of Storage Facility 1`] = `
<div>
  <div
    class="create-storage-document css-1l99ixu-OuterContainer e4u1u2g0"
  >
    <div
      class="css-n57fe4-InnerContainer e4u1u2g1"
    >
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <div
          class="css-1w80l5h-StyledColumn e4u1u2g0"
        >
          <a
            class="e1vcssft0 css-zk98nb-Anchor-StyledHoc e4u1u2g0"
            href="/orchestration/api/v1/storageNotes/back?n=/create-storage-document/undefined/add-storage-facility-details/"
          >
            Back
          </a>
          <h1
            class="e1vcssft0 css-11nzzut-StyledHeader-StyledHoc e4u1u2g0"
          >
            You have added 2 storage facilities
          </h1>
        </div>
      </div>
      <form
        action="/orchestration/api/v1/storageNotes/saveAndValidate?c=/create-storage-document/undefined/you-have-added-a-storage-facility"
        method="POST"
      >
        <h2
          class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
        >
          Facility name
        </h2>
        <dl
          class="css-39qxwj"
        >
          <div
            class="rowWithValidation css-16k4qo5"
            id="storageFacilities-0"
          >
            <dt
              class="css-a1pm7p"
              id="storageFacilities-0-facilityName"
            >
              <span
                class=""
              >
                Facility 1
              </span>
            </dt>
            <span
              class="css-3ay1af"
            >
              <a
                class="css-80kuzi-StyledHoc e109g54z0"
                href="/create-storage-document/undefined/add-storage-facility-details/0"
                id="edit-facility-0"
              >
                Edit
                <span
                  class="govuk-visually-hidden"
                >
                  storage facility Facility 1
                </span>
              </a>
            </span>
            <span
              class="css-3ay1af"
            >
              <button
                class="css-17uw1q2-StyledButton-StyledHoc e4u1u2g0"
                formaction="/orchestration/api/v1/storageNotes/removeKey?n=/create-storage-document/undefined/add-storage-facility-details&key=storageFacilities.0"
                id="remove-facility-0"
              >
                Remove
                <span
                  class="govuk-visually-hidden"
                >
                  storage facility Facility 1
                </span>
              </button>
            </span>
          </div>
        </dl>
        <dl
          class="css-39qxwj"
        >
          <div
            class="rowWithValidation css-16k4qo5"
            id="storageFacilities-1"
          >
            <dt
              class="css-a1pm7p"
              id="storageFacilities-1-facilityName"
            >
              <span
                class=""
              >
                Facility 2
              </span>
            </dt>
            <span
              class="css-3ay1af"
            >
              <a
                class="css-80kuzi-StyledHoc e109g54z0"
                href="/create-storage-document/undefined/add-storage-facility-details/1"
                id="edit-facility-1"
              >
                Edit
                <span
                  class="govuk-visually-hidden"
                >
                  storage facility Facility 2
                </span>
              </a>
            </span>
            <span
              class="css-3ay1af"
            >
              <button
                class="css-17uw1q2-StyledButton-StyledHoc e4u1u2g0"
                formaction="/orchestration/api/v1/storageNotes/removeKey?n=/create-storage-document/undefined/add-storage-facility-details&key=storageFacilities.1"
                id="remove-facility-1"
              >
                Remove
                <span
                  class="govuk-visually-hidden"
                >
                  storage facility Facility 2
                </span>
              </button>
            </span>
          </div>
        </dl>
        <br />
        <br />
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <div
            class="css-1w80l5h-StyledColumn e4u1u2g0"
          >
            <h2
              class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
            >
              Do you need to add another storage facility?
            </h2>
            <div
              class="e1vcssft0 css-1phgi8s-StyledFieldset-StyledHoc e4u1u2g0"
              id="addAnotherStorageFacility"
            >
              <span
                class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
              />
              <fieldset>
                <legend
                  class="visually-hidden"
                >
                  Do you need to add another storage facility?
                </legend>
                <label
                  class="css-y8ql23-StyledHoc"
                  for="addAnotherStorageFacilityYes"
                  id="label-addAnotherStorageFacilityYes"
                >
                  <input
                    class="css-1xt3i3l"
                    id="addAnotherStorageFacilityYes"
                    name="addAnotherStorageFacility"
                    type="radio"
                    value="Yes"
                  />
                  <span
                    class="css-1uwxoxa"
                  >
                    Yes
                  </span>
                </label>
                <label
                  class="css-y8ql23-StyledHoc"
                  for="addAnotherStorageFacilityNo"
                  id="label-addAnotherStorageFacilityNo"
                >
                  <input
                    class="css-1xt3i3l"
                    id="addAnotherStorageFacilityNo"
                    name="addAnotherStorageFacility"
                    type="radio"
                    value="No"
                  />
                  <span
                    class="css-1uwxoxa"
                  >
                    No
                  </span>
                </label>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <button
            class="css-17uw1q2-StyledButton-StyledHoc e4u1u2g0"
            formaction="/orchestration/api/v1/storageNotes/saveAndValidate?saveAsDraftUrl=undefined&c=/create-storage-document/undefined/you-have-added-a-storage-facility"
            id="saveAsDraft"
            name="saveAsDraft"
            type="submit"
          >
            Save as Draft
          </button>
          <input
            name="journey"
            type="hidden"
            value="storageNotes"
          />
          <input
            name="currentUri"
            type="hidden"
            value=""
          />
          <button
            class="css-1gc9650-StyledButton-StyledHoc e4u1u2g0"
            id="continue"
            type="submit"
          >
            Save and continue
          </button>
        </div>
      </form>
      <a
        href="/create-storage-document/progress"
        id="backToProgress"
      >
        Back to Your Progress
        <span
          class="govuk-visually-hidden"
        >
          back to your progress page link
        </span>
      </a>
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <div
          class="css-1w80l5h-StyledColumn e4u1u2g0"
        >
          <hr
            class="help-link"
          />
          <h2
            class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
          >
            Need help?
          </h2>
          <a
            href="undefined"
            rel="noopener noreferrer"
            target="_blank"
          >
            Get help exporting fish from the UK (gov.uk)
             
            <span
              class="govuk-visually-hidden"
            >
              (opens in new tab)
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should take snapshot of the page based on number of facilities added should take a snapshot of Storage Facility when no storage facilities added 1`] = `
<div>
  <div
    class="create-storage-document css-1l99ixu-OuterContainer e4u1u2g0"
  >
    <div
      class="css-n57fe4-InnerContainer e4u1u2g1"
    >
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <div
          class="css-1w80l5h-StyledColumn e4u1u2g0"
        >
          <a
            class="e1vcssft0 css-zk98nb-Anchor-StyledHoc e4u1u2g0"
            href="/orchestration/api/v1/storageNotes/back?n=/create-storage-document/undefined/add-products-to-this-export"
          >
            Back
          </a>
          <h1
            class="e1vcssft0 css-11nzzut-StyledHeader-StyledHoc e4u1u2g0"
          >
            You have added 0 storage facility
          </h1>
        </div>
      </div>
      <form
        action="/orchestration/api/v1/storageNotes/saveAndValidate?c=/create-storage-document/undefined/you-have-added-a-storage-facility"
        method="POST"
      >
        <h2
          class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
        >
          Facility name
        </h2>
        <br />
        <br />
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <div
            class="css-1w80l5h-StyledColumn e4u1u2g0"
          >
            <h2
              class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
            >
              Do you need to add another storage facility?
            </h2>
            <div
              class="e1vcssft0 css-1phgi8s-StyledFieldset-StyledHoc e4u1u2g0"
              id="addAnotherStorageFacility"
            >
              <span
                class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
              />
              <fieldset>
                <legend
                  class="visually-hidden"
                >
                  Do you need to add another storage facility?
                </legend>
                <label
                  class="css-y8ql23-StyledHoc"
                  for="addAnotherStorageFacilityYes"
                  id="label-addAnotherStorageFacilityYes"
                >
                  <input
                    class="css-1xt3i3l"
                    id="addAnotherStorageFacilityYes"
                    name="addAnotherStorageFacility"
                    type="radio"
                    value="Yes"
                  />
                  <span
                    class="css-1uwxoxa"
                  >
                    Yes
                  </span>
                </label>
                <label
                  class="css-y8ql23-StyledHoc"
                  for="addAnotherStorageFacilityNo"
                  id="label-addAnotherStorageFacilityNo"
                >
                  <input
                    class="css-1xt3i3l"
                    id="addAnotherStorageFacilityNo"
                    name="addAnotherStorageFacility"
                    type="radio"
                    value="No"
                  />
                  <span
                    class="css-1uwxoxa"
                  >
                    No
                  </span>
                </label>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <button
            class="css-17uw1q2-StyledButton-StyledHoc e4u1u2g0"
            formaction="/orchestration/api/v1/storageNotes/saveAndValidate?saveAsDraftUrl=undefined&c=/create-storage-document/undefined/you-have-added-a-storage-facility"
            id="saveAsDraft"
            name="saveAsDraft"
            type="submit"
          >
            Save as Draft
          </button>
          <input
            name="journey"
            type="hidden"
            value="storageNotes"
          />
          <input
            name="currentUri"
            type="hidden"
            value=""
          />
          <button
            class="css-1gc9650-StyledButton-StyledHoc e4u1u2g0"
            id="continue"
            type="submit"
          >
            Save and continue
          </button>
        </div>
      </form>
      <a
        href="/create-storage-document/progress"
        id="backToProgress"
      >
        Back to Your Progress
        <span
          class="govuk-visually-hidden"
        >
          back to your progress page link
        </span>
      </a>
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <div
          class="css-1w80l5h-StyledColumn e4u1u2g0"
        >
          <hr
            class="help-link"
          />
          <h2
            class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
          >
            Need help?
          </h2>
          <a
            href="undefined"
            rel="noopener noreferrer"
            target="_blank"
          >
            Get help exporting fish from the UK (gov.uk)
             
            <span
              class="govuk-visually-hidden"
            >
              (opens in new tab)
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should take snapshot of the page based on number of facilities added should take a snapshot of Storage Facility when only 1 facility added 1`] = `
<div>
  <div
    class="create-storage-document css-1l99ixu-OuterContainer e4u1u2g0"
  >
    <div
      class="css-n57fe4-InnerContainer e4u1u2g1"
    >
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <div
          class="css-1w80l5h-StyledColumn e4u1u2g0"
        >
          <a
            class="e1vcssft0 css-zk98nb-Anchor-StyledHoc e4u1u2g0"
            href="/orchestration/api/v1/storageNotes/back?n=/create-storage-document/undefined/add-storage-facility-details/"
          >
            Back
          </a>
          <h1
            class="e1vcssft0 css-11nzzut-StyledHeader-StyledHoc e4u1u2g0"
          >
            You have added 1 storage facility
          </h1>
        </div>
      </div>
      <form
        action="/orchestration/api/v1/storageNotes/saveAndValidate?c=/create-storage-document/undefined/you-have-added-a-storage-facility"
        method="POST"
      >
        <h2
          class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
        >
          Facility name
        </h2>
        <dl
          class="css-39qxwj"
        >
          <div
            class="rowWithValidation css-16k4qo5"
            id="storageFacilities-0"
          >
            <dt
              class="css-a1pm7p"
              id="storageFacilities-0-facilityName"
            >
              <span
                class=""
              >
                Facility 1
              </span>
            </dt>
            <span
              class="css-3ay1af"
            >
              <a
                class="css-80kuzi-StyledHoc e109g54z0"
                href="/create-storage-document/undefined/add-storage-facility-details/0"
                id="edit-facility-0"
              >
                Edit
                <span
                  class="govuk-visually-hidden"
                >
                  storage facility Facility 1
                </span>
              </a>
            </span>
            <span
              class="css-3ay1af"
            />
          </div>
        </dl>
        <br />
        <br />
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <div
            class="css-1w80l5h-StyledColumn e4u1u2g0"
          >
            <h2
              class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
            >
              Do you need to add another storage facility?
            </h2>
            <div
              class="e1vcssft0 css-1phgi8s-StyledFieldset-StyledHoc e4u1u2g0"
              id="addAnotherStorageFacility"
            >
              <span
                class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
              />
              <fieldset>
                <legend
                  class="visually-hidden"
                >
                  Do you need to add another storage facility?
                </legend>
                <label
                  class="css-y8ql23-StyledHoc"
                  for="addAnotherStorageFacilityYes"
                  id="label-addAnotherStorageFacilityYes"
                >
                  <input
                    class="css-1xt3i3l"
                    id="addAnotherStorageFacilityYes"
                    name="addAnotherStorageFacility"
                    type="radio"
                    value="Yes"
                  />
                  <span
                    class="css-1uwxoxa"
                  >
                    Yes
                  </span>
                </label>
                <label
                  class="css-y8ql23-StyledHoc"
                  for="addAnotherStorageFacilityNo"
                  id="label-addAnotherStorageFacilityNo"
                >
                  <input
                    class="css-1xt3i3l"
                    id="addAnotherStorageFacilityNo"
                    name="addAnotherStorageFacility"
                    type="radio"
                    value="No"
                  />
                  <span
                    class="css-1uwxoxa"
                  >
                    No
                  </span>
                </label>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <button
            class="css-17uw1q2-StyledButton-StyledHoc e4u1u2g0"
            formaction="/orchestration/api/v1/storageNotes/saveAndValidate?saveAsDraftUrl=undefined&c=/create-storage-document/undefined/you-have-added-a-storage-facility"
            id="saveAsDraft"
            name="saveAsDraft"
            type="submit"
          >
            Save as Draft
          </button>
          <input
            name="journey"
            type="hidden"
            value="storageNotes"
          />
          <input
            name="currentUri"
            type="hidden"
            value=""
          />
          <button
            class="css-1gc9650-StyledButton-StyledHoc e4u1u2g0"
            id="continue"
            type="submit"
          >
            Save and continue
          </button>
        </div>
      </form>
      <a
        href="/create-storage-document/progress"
        id="backToProgress"
      >
        Back to Your Progress
        <span
          class="govuk-visually-hidden"
        >
          back to your progress page link
        </span>
      </a>
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <div
          class="css-1w80l5h-StyledColumn e4u1u2g0"
        >
          <hr
            class="help-link"
          />
          <h2
            class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
          >
            Need help?
          </h2>
          <a
            href="undefined"
            rel="noopener noreferrer"
            target="_blank"
          >
            Get help exporting fish from the UK (gov.uk)
             
            <span
              class="govuk-visually-hidden"
            >
              (opens in new tab)
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
